src/System.Management.Automation/engine/TypeTable.cs

```diff
@@ -3392,6 +3394,7 @@ internal TypeTable(bool isShared)
         {
             this.isShared = isShared;
             _typeFileList = new List<string>();
+            _memberFactoryFunc = MemberFactory;
```

Comment:
Is the filed `_memberFactoryFunc` really needed? Why not directly use `_consolidatedMembers.GetOrAdd(types.Key, MemberFactory, types)` instead?

Reply:
Got it. You are trying to avoid constructing a `Func<...>` object every time running this method. Nice thought!

There are 4 constructors for `TypeTable`, and adding `_memberFactoryFunc = MemberFactory` here only covers 2 of them, can you add this initialization to the other constructors?

Reply:
Also, if would be helpful to add a comment to the `_memberFactoryFunc` field to explain its purpose. If you cannot get to this PR in the short term, please let me know, I'm happy to make these minor changes for you.

Reply:
@daxian-dbw Done. Yes, I profiled allocations and found that this was called so frequently, and generated lots of short lived delegates.

Reply:
Nice catch with the ctors - I misread the code and thought they all delegated to a single one. They do now.